openedx/core/djangoapps/oauth_dispatch/docs/decisions/0009-jwt-in-session-cookie.rst

```diff
@@ -0,0 +1,203 @@
+
+   * *aud* - should remove this since we do not make use of the audience field.
+   * *preferred_username* - should be renamed simply to *username*.
+   * *administrator* - can keep for now, but may eventually be replaced as *role* data - when we design
+     authorization.
+
+.. _HTTP Cookie RFC standard: https://tools.ietf.org/html/rfc6265
+.. _up to 4096 bytes: https://tools.ietf.org/html/rfc6265#section-6.1
+.. _Modern browsers have treated this requirement as a maximum: http://browsercookielimits.squawky.net/
+.. _currently embedded in the JWT: https://github.com/edx/edx-platform/blob/92030ea15216a6641c83dd7bb38a9b65112bf31a/openedx/core/lib/token_utils.py#L13
+
+
+JWT Cookie Security
+^^^^^^^^^^^^^^^^^^^
+
+#. **Enable CSRF Protection.** Storing JWTs in HTTP cookies is potentially vulnerable to CSRF attacks.
+   See `JWT Cookie Storage Security`_. To protect against this:
+   
+   * Enable the HttpOnly_ flag on the **"JWT Signature Cookie"**, so Javascript code cannot misuse the JWT.
+   * Enable the Secure_ flag on the cookie, so it will not be sent (and thus leaked) through an unencrypted channel.
+   * Enable `Django's CSRF middleware`_ for every response.
+   * Ensure all GET requests are side-effect free.
+   
+     * Note: The `same-origin policy`_ protects against CSRF attacks on GET requests since the rogue website cannot
+       access the response from the GET request.
+     * However, even though the rogue website cannot access the response, the GET request is still processed on the
+       server before returning the response. So we need to ensure there are no unwanted side-effects on the server.
+
+#. **CORS and withCredentials.** `Cross-origin resource sharing (CORS)`_ will need to be configured so that all allowed
+   microfrontends can access the necessary backend microservices. In addition, microfrontends will need to set the
+   withCredentials_ attribute so that the JWT Cookie gets sent when API calls are made.
+
+   Note: We cannot selectively choose which cookies are sent so all edX-issued cookies will be sent with these API
+   calls. Apparently, we already send all edX cookies on API requests today, so this will not cause a significant
+   performance issue.
+
+
+.. _JWT Cookie Storage Security: https://stormpath.com/blog/where-to-store-your-jwts-cookies-vs-html5-web-storage#so-whats-the-difference
+.. _HttpOnly: https://www.owasp.org/index.php/HttpOnly
+.. _Secure: https://www.owasp.org/index.php/SecureFlag
+.. _`Django's CSRF middleware`: https://docs.djangoproject.com/en/1.11/ref/csrf/
+.. _same-origin policy: https://en.wikipedia.org/wiki/Same-origin_policy
+.. _Cross-origin resource sharing (CORS): https://en.wikipedia.org/wiki/Cross-origin_resource_sharing
+.. _withCredentials: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials
+
+
+Consequences
+------------
+
+#. Instead of storing JWTs in cookies, microfrontends could store them in HTML5 Web Storage. However, that is
```

Comment:
I find this confusing to have here. Can this move to a Rejected Alternates section?
This reminds me that I left comments on one of the POC PRs already.